core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchReaderIterator.java

```diff
@@ -616,8 +616,7 @@ private ScanServerData rebinToScanServers(Map<String,Map<KeyExtent,List<Range>>>
     // get a snapshot of this once,not each time the plugin request it
     var scanAttemptsSnapshot = scanAttempts.snapshot();
 
-    Duration timeoutLeft = Duration.ofMillis(retryTimeout)
-        .minus(Duration.ofMillis(System.currentTimeMillis() - startTime));
+    Duration timeoutLeft = Duration.ofMillis(retryTimeout).minus(startTime.elapsed());
```

Comment:
May be able to do something like the following, do not think this will compile

```suggestion
    Duration timeoutLeft = Duration.ofMillis(retryTimeout - startTime.elapsed(MILLIS));
```